src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/ObjectiveC/MessageSendTests.cs

```diff
@@ -16,6 +16,7 @@ namespace System.Runtime.InteropServices.Tests
 {
     [PlatformSpecific(TestPlatforms.OSX)]
     [SkipOnMono("Not currently implemented on Mono")]
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsNotNativeAot))]
```

Comment:
```suggestion
    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsNotNativeAot))] // https://github.com/dotnet/runtimelab/issues/155
```

Reply:
Funnily enough that's what got me started month ago, now laying somewhere deep in my git stash 😅

Reply:
It should not be too hard to implement this one.

Reply:
It's not hard. For the reference tracking API the tricky part is to do it efficiently so the price is paid only for tracked objects. I had some prototype working but it had a lot of rough edges.